### PR TITLE
Allow fine tuning of the label's position relative to it's calculated position.

### PR DESCRIPTION
**Pull request by plexus77**

### CODE REVIEW

**Comment by neveldo** [comment]
Hello,

Thank for the PR !! getTextPosition() is called two times (one for the text initialization, where you have added you code, and another for the text update, that can occur when a user triggers an 'update' event on the container). Hence, I think that this feature should take place into the getTextPosition() in order to factorize the code.

Maybe, we could make the argument margin of the function getTextPosition() accept an integer (as currently) or an object such as `{'x' : 5, 'y' : 10}` that would allow us to fine tune the vertical and horizontal margin if needed.

If we receive an integer, we could transform the margin argument to an object : 

```javascript
if (typeof margin === "number") {
    margin = {x: margin, y: margin};
}
```

And then, we can deal with this object in the rest of  the function to compute the proper text position.

What do you think about it ?

File: js/jquery.mapael.js [hunk 1]
```diff
@@ -358,7 +358,15 @@
                 bbox = elem.mapElem.getBBox();
                 textPosition = self.getTextPosition(bbox, elemOptions.text.position, elemOptions.text.margin);
                 elemOptions.text.attrs["text-anchor"] = textPosition.textAnchor;
-                elem.textElem = self.paper.text(textPosition.x, textPosition.y, elemOptions.text.content).attr(elemOptions.text.attrs);
+
+                // Allow fine tuning of the label's position relative to it's calculated position.
+                var dx = 0, dy = 0;
+                if(typeof options.text.attrs.dx !== undefined) dx = options.text.attrs.dx;
```
Comment (Indigo744) [inline js/jquery.mapael.js]: It should be
```javascript
if(options.text.attrs.dx !== undefined)
```
Note: if you are using `typeof`, you should compare to `'undefined'` (notice the quote! `typeof` returns a string).
For Mapael, you can directly use `undefined`.

File: js/jquery.mapael.js [hunk 2]
```diff
@@ -358,7 +358,15 @@
...
+                if(typeof options.text.attrs.dy !== undefined) dy = options.text.attrs.dy;
```
Comment (Indigo744) [inline js/jquery.mapael.js]: Same as above.

**Comment by Indigo744** [comment]
I agree with @neveldo. dx/dy should be set inside `getTextPosition()` function.

Regarding your second point, I think the `margin` name is not the best since we are not setting a real margin: we are adding a *shift* (or delta) to the position `(x, y)`. Hence I think `(dx, dy)` fits nicely the description.

**Comment by plexus77** [comment]
Hi guys,
I only just saw this thread sorry, I would have refactored, let me know if you still need me to do anything.
Cheers :)

**Comment by neveldo** [comment]
Hello @plexus77 ,

Thank you, don't worry, we are working on it :)

**Pull request closed without merging**